src/main/plugins/org.dita.pdf2/xsl/fo/tables.xsl

```diff
@@ -1042,17 +1090,25 @@ See the accompanying LICENSE file for applicable license.
     </xsl:template>
 
     <xsl:template match="*[contains(@class, ' topic/strow ')]/*[contains(@class, ' topic/stentry ')]">
+        <xsl:param name="number-cells" as="xs:integer" tunnel="yes">
+          <xsl:apply-templates select="../.." mode="count-max-simpletable-cells"/>
+        </xsl:param>
         <fo:table-cell xsl:use-attribute-sets="strow.stentry">
             <xsl:call-template name="commonattributes"/>
-            <xsl:variable name="entryCol" select="count(preceding-sibling::*[contains(@class, ' topic/stentry ')]) + 1"/>
+            <xsl:call-template name="simpletableApplySpansAttrs"/>
+            <xsl:variable name="entryCol" select="@dita-ot:x"/>
             <xsl:variable name="frame" as="xs:string" select="(ancestor::*[contains(@class, ' topic/simpletable ')][1]/@frame, $table.frame-default)[1]"/>
+            <xsl:variable name="last-column" as="xs:integer">
+              <xsl:variable name="span" select="if (exists(@colspan)) then (xs:integer(@colspan) - 1) else 0"/>
```

Comment:
(Maybe a thought for the future) Should we be doing validation of `@colspan` as integer, or do we do that earlier? I think using `xs:integer(@colspan)` will throw messy errors if it's not an integer - I've seen that with code that assumed an integer with proportional column widths, and failed when it got `.8*` instead of `8*` ... but that seems less likely here

Reply:
We have clean-up filters that we could use to implement the integer validation. I'll leave it for future.